plonky2/src/plonk/proof.rs

```diff
@@ -81,9 +81,10 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
 {
     pub fn compress(
         self,
+        circuit_digest: &<<C as GenericConfig<D>>::Hasher as Hasher<C::F>>::Hash,
```

Comment:
What do you think of having this (and `decompress`) take `VerifierCircuitData`? To combine the two params into one.

Or we could do it as a larger refactor later. Seems like we have a lot of code where `VerifierOnlyCircuitData`/`CommonCircuitData` are separated, like the `recursive_verifier` test code, when combining them could make things more succinct.

Reply:
Oh, maybe that wouldn't be ideal since `VerifierCircuitData` owns its data, so it might require some cloning.

Reply:
> Oh, maybe that wouldn't be ideal since `VerifierCircuitData` owns its data, so it might require some cloning.

Yeah that's the issue I ran into here and elsewhere. There's no way currently to go from `CircuitData` to `VerifierCircuitData` without throwing away `ProverOnlyCircuitData` or some cloning. Since the prover should be able to run these functions without throwing away the prover key, I thought it would just be easier to pass the circuit digest, even though it's a bit ugly.